islandora_compound_object.module

```diff
@@ -613,41 +630,53 @@ function islandora_compound_object_retrieve_compound_info(AbstractObject $object
   $rels_predicate = variable_get('islandora_compound_object_relationship', 'isConstituentOf');
   $part_of = $object->relationships->get(FEDORA_RELS_EXT_URI, $rels_predicate);
   $info = array();
-  // This object is part of a compound object.
+
+  // This object is the child of a compound object.
   if (!empty($part_of)) {
     // Go through all parents.
+    // XXX: Handle the case where we are a member of multiple compounds?
     foreach ($part_of as $part) {
       $parent_pid = $part['object']['value'];
       $parent = islandora_object_load($parent_pid);
```

Comment:
Your comment on line 637 is a good point? This could conceivably load a lot of objects and we only use the last one...right?  If we are going to grab one parent at random, would we be better to just pop the last value off the array and exit quickly?